File: src/lcel.ts
import { config } from 'dotenv';
import { ChatOpenAI } from "@langchain/openai";
import { ChatPromptTemplate } from "@langchain/core/prompts";

// .envから環境変数を読み込む
config();

async function main() {
  // チェーンの準備
  const model = new ChatOpenAI();
  const prompt = ChatPromptTemplate.fromTemplate("{topic}についてジョークを言ってください");
  const chain = prompt.pipe(model);

  // ストリーム
  console.log("ストリーム結果:\n");
  const stream = await chain.stream({topic: "人工知能"});
  for await (const chunk of stream) {
    process.stdout.write(chunk.content);
  }
  console.log("\n");

  // invoke
  const result = await chain.invoke({topic: "ロボット"});
  console.log("Invoke結果:\n", result.content);
  console.log("\n");

  // batch
  console.log("Batch結果:\n");
  const batchResults = await chain.batch([
    {topic: "人工知能"},
    {topic: "ロボット"}
  ]);

  batchResults.forEach((result, i) => {
    console.log(`結果 ${i+1}: ${result.content}`);
  });
}

main().catch(console.error);

File: python/lcel.py
from langchain_core.prompts import ChatPromptTemplate
from langchain_openai import ChatOpenAI
from dotenv import load_dotenv

load_dotenv()

# チェーンの準備
model = ChatOpenAI()
prompt = ChatPromptTemplate.from_template("{topic}についてジョークを言ってください")
chain = prompt | model

# ストリーム
print("ストリーム結果:\n")
for s in chain.stream({"topic": "人工知能"}):
    print(s.content, end="", flush=True)

print("\n")

result = chain.invoke({"topic": "ロボット"})
print("Invoke結果:\n", result.content)  # .content でテキスト内容を取得

print("\n")

# batch の結果を表示
batch_results = chain.batch([{"topic": "人工知能"}, {"topic": "ロボット"}])
print("\nBatch結果:\n")
for i, result in enumerate(batch_results):
    print(f"結果 {i+1}: {result.content}")
